pkg/controller/infinispan/xsite.go

```diff
@@ -48,7 +48,16 @@ func ComputeXSite(infinispan *ispnv1.Infinispan, kubernetes *kube.Kubernetes, se
 
 		remoteLocations := findRemoteLocations(xsite.Name, infinispan)
 		for _, remoteLocation := range remoteLocations {
-			err := appendRemoteLocation(infinispan, &remoteLocation, kubernetes, logger, xsite)
+			var err error
+			if remoteLocation.Host != "" {
```

Comment:
@timothymathison what do you think about remote host validation. Once added, no chance to replace it without whole Infinispan CR remove/add operation. Incorrect host will display inside Infinispan server logs, not o the operator side. Not sure that's possible to validate whole host:port combination but need to be validated.

Reply:
Yeah I'm not sure what the validation would look like exactly. We could simply validate that the host can be resolved ([as is currently done](https://github.com/infinispan/infinispan-operator/blob/af542c4e456b42a962b1a07fe62f8559c23ec784/pkg/controller/infinispan/xsite.go#L217)) or that `host`/`port` can be connected to at the socket layer.
However, with multiple sites this may create a bit of a race condition: all the sites would need to be created at exactly the same time so that the site services all exist by the time each operator reaches this logic. At least that's my understanding?

Reply:
>Once added, no chance to replace it without whole Infinispan CR remove/add operation.

Does infinispan not respect later updates to the `infinispan.yaml` configuration file?

Reply:
> Does infinispan not respect later updates to the `infinispan.yaml` configuration file?

Looks like yes, but need to be validated if it will reflect on the cluster or not.

Reply:
> Yeah I'm not sure what the validation would look like exactly. We could simply validate that the host can be resolved ([as is currently done](https://github.com/infinispan/infinispan-operator/blob/af542c4e456b42a962b1a07fe62f8559c23ec784/pkg/controller/infinispan/xsite.go#L217)) or that `host`/`port` can be connected to at the socket layer.
> However, with multiple sites this may create a bit of a race condition: all the sites would need to be created at exactly the same time so that the site services all exist by the time each operator reaches this logic. At least that's my understanding?

Yes, sure. But right now there is condition, that all remote server locations must be active. Probably, there's no need to add validation, but we will know about fail server connection too late from the server log only.
@pruivo what do you think about this?

Reply:
> But right now there is condition, that all remote server locations must be active.

That's a limitation in JGroups since it requires the IP/port of the remote site before starting. In the current code, the operator provides that info and it needs the other site to be up to extract the service IP/port. 

Since we are "moving" the logic from the operator to the user, IMO it is fine not to wait for the other site to be up. Of course, the user needs to check if the xsite bridge is up or not and, if the user makes a mistake, it has to change the config and restart the cluster.

Reply:
Please also pay attention, that according to docs
```
To perform cross-site replication, Infinispan Operator requires Infinispan clusters to have the same name and run in matching namespaces.
```
https://infinispan.org/infinispan-operator/master/operator.html#backup_sites
which will not validated at the customer host/port provided case. The initial JGroups connection looks like will work, but in case of fails will not work. This need to be validated.

Reply:
The way I see is, if hardcoded host/port, you are giving up on the operator to manage xsite. 
All the docs related to xsite are invalidated in that case. You don't need to create permissions, accounts, share secrets, use the same namespace... actually, you don't need an openshift cluster in the other site. You can do xsite between opensift cluster and bare metal.